docs/install/authentication-mechanisms-gcp.md

```diff
@@ -131,3 +131,121 @@ gcloud projects add-iam-policy-binding $PROJECT_ID \
 
    Note that `google-cloud-key` and `key.json` are default values expected by
    our control plane.
+
+## Troubleshooting
+
+### Workload Identity
+
+To ensure Workload Identity correctly configured for a Google Cloud Service Account (GSA), 
+and a Kubernetes Service Account (KSA), you need to check:
+
+1. If the Google Cloud Service Account has the desired `iam-policy-binding`.
+   
+   By running the following command: 
+   ```shell
+   gcloud iam service-accounts get-iam-policy gsa-name@$PROJECT_ID.iam.gserviceaccount.com
+   ```
+   if the `iam-policy-binding` is correct, you'll find a `iam-policy-binding` similar to:
+   ```shell
+    bindings:
+    - members:
+      - serviceAccount:$PROJECT_ID.svc.id.goog[ksa-namespace/ksa-name]
+      role: roles/iam.workloadIdentityUser
+   ```
+ 2. If the Kubernetes Service Account has the desired annotation.
+   
+    By running the following command: 
+    ```shell
+    kubectl get serviceaccount ksa-name -o yaml -n ksa-namespace
+    ```
+    if the annotation exists, you'll find an annotation similar to:
+    ```shell
+    metadata:
+      annotations:
+        iam.gke.io/gcp-service-account: gsa-name@$PROJECT_ID.iam.gserviceaccount.com
+    ```
+
+### Kubernetes Secrets
+
+To ensure kubernetes Secret correctly configured in a namespace, you need to check if a Kubernetes Secret 
+named `google-cloud-key` is in desired namespace by running the following command: 
```

Comment:
```suggestion
To ensure kubernetes Secret correctly configured in a namespace, you need to check if a Kubernetes Secret 
with the correct name is in the namespace. By default, this secret is named `google-cloud-key`, but it could
be set to something else either in the custom object itself (e.g. the Source's `spec.secret`) or in the `config-gcp-auth` ConfigMap. Verify that a secret with the correct name is in the same namespace as the custom object by 
running the following command:
```